src/oph/ehoks/virkailija/handler.clj

```diff
@@ -116,23 +116,34 @@
         (assoc ::audit/changes {:new hoks}
                ::audit/target  (audit/hoks-target-data hoks-db)))))
 
-(defn- get-hoks-perustiedot
-  "Get basic information from HOKS"
-  [oppija-oid]
-  (if-let [hoksit (db-hoks/select-hoks-by-oppija-oid oppija-oid)]
-    (assoc (restful/ok hoksit)
-           ::audit/target
-           {:hoks-ids            (map :id hoksit)
-            :oppija-oid          (:oppija-oid (first hoksit))
-            :opiskeluoikeus-oids (map :opiskeluoikeus-oid hoksit)})
-    (response/not-found {:message "HOKS not found"})))
-
 (defn- any-hoks-has-active-opiskeluoikeus?
   "Check if any of the HOKSes has an active opiskeluoikeus"
   [hoksit]
   (some #(some-> (koski/get-opiskeluoikeus! %) opiskeluoikeus/active?)
         (map :opiskeluoikeus-oid hoksit)))
 
+(defn- get-oppijan-hoksit-for-virkailija
+  "Get basic information from HOKS"
+  [oppija-oid ticket-user]
+  (if-let [hoksit (db-hoks/select-hoks-by-oppija-oid oppija-oid)]
+    (let [virkailijan-oppilaitosten-hoksit
+          (filter #(user/has-privilege-to-hoks? % ticket-user :read)
+                  hoksit)
+          virkailijan-oppilaitosten-opiskeluoikeuksia-active?
+          (any-hoks-has-active-opiskeluoikeus? virkailijan-oppilaitosten-hoksit)
```

Comment:
Ehdotus: Tää päättely vois olla hyvä eriyttää johonkin omaan funktioonsa kuten esim. `oppija-has-active-oo-in-virkailijas-organisations?`.  Huomaan kyllä et virkailijan-oppilaitosten-hoksit-tietoa käytetään tuossa alempana... miten vaan siistimpi.

Reply:
Totta, joo yksinkertaistin tätä hieman. Toteutus on nyt aika lähellä vähän alempana olevaa `get-oppilaitoksen-oppijan-hoksit`, mutta tässä kohtaa kuitenkin halutaan tehdä suodatus virkailijan kaikkien laitosten hoksien perusteella.

Tuosta lienee myöhemmin helpompi lähteä liikkeelle tekemään sitä isompaa käyttöoikeusremppaa.